samples/headless-ssr/commerce-generic/src/components/ProductGrid.ts

```diff
@@ -0,0 +1,49 @@
+import type {Product, ProductList} from '@coveo/headless/ssr-commerce-next';
+import {getElement} from '../common/utils.js';
+
+export function ProductGrid(productList: ProductList) {
+  if (!productList) return;
+
+  const grid = getElement<HTMLDivElement>('product-grid');
+  const noProducts = getElement<HTMLDivElement>('no-products');
+  if (!grid) return;
+
+  const render = () => {
+    const products = selectProducts(productList);
+    const hasProducts = products.length > 0;
+
+    if (noProducts) {
+      noProducts.style.display = hasProducts ? 'none' : 'block';
+    }
+    grid.innerHTML = hasProducts ? renderProductGrid(products) : '';
+  };
+
+  productList.subscribe(render);
+  render();
+}
+
+export function renderProductCard(product: Product): string {
+  const imageUrl = product.ec_thumbnails?.[0] ?? product.ec_images?.[0] ?? '';
+  const name = product.ec_name ?? 'Unknown Product';
+  const brand = product.ec_brand ?? '';
+  const price = product.ec_price ?? product.ec_promo_price ?? '';
+  const rating = product.ec_rating ?? 0;
+
+  return `
```

Comment:
Eventually we should log an ec_productClick event when a product card is clicked, but this can be added later.